File: backend/app/models/platform_config.py
"""
Platform Configuration model for managing all application settings
"""
from sqlalchemy import Column, String, DateTime, Boolean, JSON, Text, Enum as SQLEnum, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid
from datetime import datetime
import enum
from app.core.database import Base


class ConfigCategory(str, enum.Enum):
    """Configuration categories"""
    APPLICATION = "application"
    SECURITY = "security"
    DATABASE = "database"
    REDIS = "redis"
    QDRANT = "qdrant"
    OPENAI = "openai"
    FILE_STORAGE = "file_storage"
    API = "api"
    CORS = "cors"
    RATE_LIMITING = "rate_limiting"
    LOGGING = "logging"


class ConfigValueType(str, enum.Enum):
    """Configuration value types"""
    STRING = "string"
    INTEGER = "integer"
    BOOLEAN = "boolean"
    JSON = "json"
    SECRET = "secret"  # For passwords, API keys, tokens


class PlatformConfiguration(Base):
    """Platform-wide configuration settings"""
    __tablename__ = "platform_configurations"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    
    # Configuration key (e.g., "SECRET_KEY", "CORS_ORIGINS", "MAX_UPLOAD_SIZE")
    config_key = Column(String(100), unique=True, nullable=False, index=True)
    
    # Configuration category
    category = Column(SQLEnum(ConfigCategory), nullable=False, index=True)
    
    # Value type
    value_type = Column(SQLEnum(ConfigValueType), nullable=False)
    
    # Configuration value (JSON for complex types, string for simple)
    # For SECRET type, value is encrypted/hashed
    config_value = Column(Text, nullable=True)  # Store as JSON string for complex types
    
    # Display value (for secrets, this is masked)
    display_value = Column(String(500), nullable=True)  # Masked version for secrets
    
    # Metadata
    description = Column(Text, nullable=True)  # Human-readable description
    is_secret = Column(Boolean, default=False, nullable=False)  # Whether this is a secret
    is_required = Column(Boolean, default=False, nullable=False)  # Whether this is required
    is_encrypted = Column(Boolean, default=False, nullable=False)  # Whether value is encrypted
    
    # Change tracking
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    updated_by = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    creator = relationship("User", foreign_keys=[created_by])
    updater = relationship("User", foreign_keys=[updated_by])

File: frontend/src/lib/platformConfig.ts
import { api } from './api'

export interface PlatformConfig {
  id: string
  config_key: string
  category: string
  value_type: string
  value: any
  display_value?: string
  description?: string
  is_secret: boolean
  is_required: boolean
  created_at: string
  updated_at: string
}

export interface PlatformConfigCreate {
  config_key: string
  value: any
  category: string
  value_type: string
  description?: string
  is_secret: boolean
}

export interface PlatformConfigUpdate {
  value: any
  description?: string
}

export const platformConfigApi = {
  list: async (category?: string): Promise<PlatformConfig[]> => {
    const params: any = {}
    if (category) params.category = category
    console.log('Fetching platform configs with params:', params)
    const response = await api.get('/platform-config', { params })
    console.log('Platform configs API response:', response.data)
    console.log('Response length:', response.data?.length ?? 0)
    return response.data
  },

  get: async (configKey: string): Promise<PlatformConfig> => {
    const response = await api.get(`/platform-config/${configKey}`)
    return response.data
  },

  create: async (config: PlatformConfigCreate): Promise<PlatformConfig> => {
    const response = await api.post('/platform-config', config)
    return response.data
  },

  update: async (configKey: string, config: PlatformConfigUpdate): Promise<PlatformConfig> => {
    const response = await api.put(`/platform-config/${configKey}`, config)
    return response.data
  },

  delete: async (configKey: string): Promise<void> => {
    await api.delete(`/platform-config/${configKey}`)
  },

  getCategories: async (): Promise<string[]> => {
    const response = await api.get('/platform-config/categories/list')
    return response.data
  },

  getValueTypes: async (): Promise<string[]> => {
    const response = await api.get('/platform-config/value-types/list')
    return response.data
  },
}

